src/storage/src/sink/kafka.rs

```diff
@@ -354,12 +413,18 @@ impl TransactionalProducer {
             key: "materialize-timestamp",
             value: Some(time.to_string().as_bytes()),
         });
+        let partition = match self.partition_strategy {
+            SinkPartitionStrategy::Legacy => None,
+            SinkPartitionStrategy::Stable => {
+                Some(i32::try_from(key_hash % self.partition_count).unwrap())
```

Comment:
This has a small defect that the current code doesn't have: the number of partitions does not dynamically update. librdkafka's default partitioner will periodically refresh the topic metadata in the background, discover new partitions, and use them in its partitioner. 

We may want to investigate using `rd_kafka_topic_conf_set_partitioner_cb` to set a partition callback, which is invoked to determine the partition of every message gets librdkafka's current knowledge of the partition count. That'd be a substantial replumbing of the hashing logic though.

So maybe this is just fine for now. We could always add our own background task in the future, like we have for consumers, that periodically refreshes the topic's metadata and updates `self.partition_count`.

Reply:
Ah, of course, the opaque! Makes sense.